tt_metal/impl/dispatch/kernels/cq_dispatch.cpp

```diff
@@ -381,27 +381,33 @@ void process_write_linear(uint32_t num_mcast_dests) {
     uint32_t dst_addr = cmd->write_linear.addr + write_offset[write_offset_index];
     uint32_t length = cmd->write_linear.length;
     uint32_t data_ptr = cmd_ptr + sizeof(CQDispatchCmd);
+    cq_noc_async_write_init_state<CQ_NOC_sNdl, multicast>(0, get_noc_addr_helper(dst_noc, dst_addr));
+
+    uint32_t writes = 0;
+    uint32_t mcasts = noc_nonposted_writes_acked[noc_index];
+
     while (length != 0) {
         uint32_t xfer_size = (length > dispatch_cb_page_size) ? dispatch_cb_page_size : length;
-        uint64_t dst = get_noc_addr_helper(dst_noc, dst_addr);
         // "Reserve" pages for the next write from this block
-        block_noc_writes_to_clear[rd_block_idx]++;
+        writes++;
+        mcasts += num_mcast_dests;
         // Get a page if needed
         if (data_ptr + xfer_size > cb_fence) {
             // Check for block completion
             if (cb_fence == block_next_start_addr[rd_block_idx]) {
                 uint32_t orphan_size = cb_fence - data_ptr;
                 // No more writes from this block. Decrement the number of writes
                 // since they were all accounted for.
-                block_noc_writes_to_clear[rd_block_idx] -= (orphan_size == 0);
+                writes -= (orphan_size == 0);
+                mcasts -= (orphan_size == 0) * num_mcast_dests;
```

Comment:
These should get removed I think once we move to writing up to fence instead of 4k transactions. Currently I just copied the impl from our other fns. Perf ci is passing though havent looked at actual perf differences yet.

Reply:
Removed on this branch: https://github.com/tenstorrent/tt-metal/tree/asaigal/remove_orphan_writes.